src/examples/zkapps/reducer/run-live.ts

```diff
@@ -0,0 +1,143 @@
+}
+
+tic('Run reducer examples against real network.');
+console.log();
+const network = Mina.Network({
+  mina: useCustomLocalNetwork
+    ? 'http://localhost:8080/graphql'
+    : 'https://berkeley.minascan.io/graphql',
+  archive: useCustomLocalNetwork
+    ? 'http://localhost:8282'
+    : 'https://api.minascan.io/archive/berkeley/v1/graphql',
+  lightnetAccountManager: 'http://localhost:8181',
+});
+Mina.setActiveInstance(network);
+
+let { keys, addresses } = randomAccounts('contract', 'user1', 'user2');
+
+let tx, pendingTx: Mina.PendingTransaction, balances, oldBalances;
+
+// compile contracts & wait for fee payer to be funded
+const senderKey = useCustomLocalNetwork
+  ? (await Lightnet.acquireKeyPair()).privateKey
+  : PrivateKey.random();
+const sender = senderKey.toPublicKey();
+
+const sender2Key = useCustomLocalNetwork
+  ? (await Lightnet.acquireKeyPair()).privateKey
+  : PrivateKey.random();
+const sender2 = sender2Key.toPublicKey();
+
+tic('Compiling Merkle List Reducer Smart Contract');
+await MerkleListReducing.compile();
+toc();
+const merkleListReducerContract = new MerkleListReducing(addresses.contract);
+
+let senderSpec = { sender, fee: 1000000000 };
+let sender2Spec = { sender: sender2, fee: 1000000000 };
+
+tic('deploy contracts');
+let deployTx = await Mina.transaction(senderSpec, async () => {
+  AccountUpdate.createSigned(sender).balance.subInPlace(
+    Mina.getNetworkConstants().accountCreationFee
+  );
+  await merkleListReducerContract.deploy();
+});
+pendingTx = await deployTx.sign([senderKey, keys.contract]).send();
+
+await pendingTx.wait();
+toc();
+console.log('working so far');
+// push some actions
+
+tic('dispatch transactions');
+let dispatchTx = await Mina.transaction(senderSpec, async () => {
+  await merkleListReducerContract.postAddress(addresses.user1);
+  await merkleListReducerContract.postAddress(addresses.contract);
+  await merkleListReducerContract.postTwoAddresses(addresses.user2, sender);
+  await merkleListReducerContract.postAddress(addresses.user2);
+  await merkleListReducerContract.postTwoAddresses(
+    addresses.contract,
+    addresses.user1
+  );
+});
+await dispatchTx.prove();
+pendingTx = await dispatchTx.sign([senderKey]).send();
+await pendingTx.wait();
+toc();
+
+tic('proving inclusion');
+// check if the actions contain the `sender` address
+let containsTx = await Mina.transaction(senderSpec, async () => {
+  await merkleListReducerContract.assertContainsAddress(sender);
+  await merkleListReducerContract.assertContainsAddress(addresses.user1);
+  await merkleListReducerContract.assertContainsAddress(addresses.user2);
+  await merkleListReducerContract.assertContainsAddress(addresses.contract);
+});
+await containsTx.prove();
+pendingTx = await containsTx.sign([senderKey]).send();
+await pendingTx.wait();
+toc();
+
+tic('dispatch transactions (multi-transaction)');
+
+tic('building');
+const txs = [];
+let dispatchTx1 = await Mina.transaction(senderSpec, async () => {
```

Comment:
In these transactions, different senders create actions to reduce within the same block